controllers/manilaapi_controller.go

```diff
@@ -638,33 +638,45 @@ func (r *ManilaAPIReconciler) reconcileNormal(ctx context.Context, instance *man
 	//
 	// check for required OpenStack secret holding passwords for service/admin user and add hash to the vars map
 	//
-	ctrlResult, err := r.getSecret(ctx, helper, instance, instance.Spec.TransportURLSecret, &configVars)
-	if err != nil {
+
+	ctrlResult, err := getServiceSecret(
+		ctx,
+		types.NamespacedName{Namespace: instance.Namespace, Name: instance.Spec.Secret},
+		[]string{
+			instance.Spec.PasswordSelectors.Service,
+		},
+		helper.GetClient(),
+		&instance.Status.Conditions,
+		manila.NormalDuration,
+		&configVars,
+	)
+	if (err != nil || ctrlResult != ctrl.Result{}) {
```

Comment:
@abays I'm curious if you agree this is equally valid to the "if... else..." logic used elsewhere. I think this compound conditional is more compact, which is a good thing given the fact that this takes place a LOT in our code. I think it would be easier on the eyes if the compound conditional like this could be used.